module/zfs/spa.c

```diff
@@ -4945,6 +4945,7 @@ spa_vdev_detach(spa_t *spa, uint64_t guid, uint64_t pguid, int replace_done)
 	 * But first make sure we're not on any *other* txg's DTL list, to
 	 * prevent vd from being accessed after it's freed.
 	 */
+	ASSERT3P(vd->vdev_path, !=, NULL);
```

Comment:
An assert seems heavy handed to me. Plus, asserts are not included in production builds so the issue would only be handled in test builds currently. `vd->vdev_path` is duplicated and only ever used in an internal log message. Maybe it would be appropriate to check for `vd->vdev_path == NULL` and print something like "<unknown>" to the internal log?